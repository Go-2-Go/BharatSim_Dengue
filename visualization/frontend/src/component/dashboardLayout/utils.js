import React from 'react';
import renderChart from '../charts/renderChart';

export function renderElement(element) {
  return (
    <div key={element.i} data-grid={element} data-testid={element.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(element.chartType, { config: element.config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
