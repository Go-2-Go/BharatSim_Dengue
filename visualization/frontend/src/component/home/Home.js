import React from 'react';
import { Box } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import ExistingUserHomeScreen from './ExistingUserHomeScreen';

function Home() {
  const history = useHistory();

  function createNewProject() {
    history.push('/project/createNew');
  }

  return (
    <Box px={32}>
      <ExistingUserHomeScreen />
    </Box>
  );
}

export default Home;
