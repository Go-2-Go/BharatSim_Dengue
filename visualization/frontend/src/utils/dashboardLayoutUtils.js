/* eslint-disable react/prop-types */

import React from 'react';
import renderChart from '../component/charts/renderChart';

export function renderElement({ layout, chartType, config }) {
  return (
    <div key={layout.i} data-grid={layout} data-testid={layout.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(chartType, { config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
