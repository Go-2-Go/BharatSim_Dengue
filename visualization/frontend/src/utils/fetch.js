import axios from 'axios';
import { httpMethods } from '../constants/fetch';

let shouldShowError;

function commonErrorHandler(err) {
  shouldShowError(true);
}

function initApiConfig({ setError }) {
  shouldShowError = setError;
}

function fetchData({ url, method = httpMethods.GET, headers, data, query, customErrorHandler }) {
  return axios({ url, method, headers, data, params: query })
    .then((res) => res.data)
    .catch((err) => {
      if (customErrorHandler) {
        customErrorHandler(err);
        return;
      }
      commonErrorHandler(err);
    });
}

function uploadData({ url, headers, data, query, customErrorHandler }) {
  return fetchData({
    url,
    method: httpMethods.POST,
    headers,
    data,
    params: query,
    customErrorHandler,
  });
}

export { fetchData, uploadData, initApiConfig };
