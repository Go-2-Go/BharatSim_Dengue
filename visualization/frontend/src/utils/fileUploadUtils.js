const fileUploadedStatus = {
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
};

function getStatusAndMessageFor(fileUploadStatus, fileName) {
  return {
    [fileUploadedStatus.ERROR]: {
      status: fileUploadedStatus.ERROR,
      message: `Error occurred while unloading ${fileName}`,
    },
    [fileUploadedStatus.SUCCESS]: {
      status: fileUploadedStatus.SUCCESS,
      message: `${fileName} successfully uploaded`,
    },
    [fileUploadedStatus.LOADING]: {
      status: fileUploadedStatus.LOADING,
      message: `uploading ${fileName}`,
    },
  }[fileUploadStatus];
}

export { fileUploadedStatus, getStatusAndMessageFor };
