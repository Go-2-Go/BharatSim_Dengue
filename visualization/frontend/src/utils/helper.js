function updateState(prevState,updatedValue){
  return Object.assign({},prevState,updatedValue)
}

export {
  updateState,
}