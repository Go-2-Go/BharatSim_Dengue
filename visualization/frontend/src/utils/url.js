const DATASOURCE_URL = '/api/dataSources';
const DASHBOARD_URL = '/api/dashboard';

export const url = {
  getDataUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}`,
  getHeaderUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}/headers`,
  DATA_SOURCES: `${DATASOURCE_URL}`,
  saveDashboard: DASHBOARD_URL,
};
