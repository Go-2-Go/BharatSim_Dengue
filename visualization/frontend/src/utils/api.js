import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async () => fetchData({ url: serviceURL.DATA_SOURCES }),

  getData: async (datasource, query) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
