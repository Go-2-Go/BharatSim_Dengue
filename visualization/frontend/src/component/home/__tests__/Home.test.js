import React from 'react';
import { render } from '@testing-library/react';
import { waitFor } from '@testing-library/dom';
import Home from '../Home';
import { api } from '../../../utils/api';

jest.mock('../../../utils/api', () => ({
  api: {
    getProjects: jest.fn(),
  },
}));

describe('Home', () => {
  it('should match snapshot for new user', async () => {
    api.getProjects.mockResolvedValue({ projects: [] });
    const { container, findByText } = render(<Home />);

    await findByText('Welcome to BharatSim');
    await waitFor(() => expect(container).toMatchSnapshot());
  });
  it('should match snapshot for existing users', async () => {
    api.getProjects.mockResolvedValue({ projects: [{ name: 'project1', _id: '1' }] });
    const { container, findByText } = render(<Home />);

    await findByText('project1');

    expect(container).toMatchSnapshot();
  });
});
