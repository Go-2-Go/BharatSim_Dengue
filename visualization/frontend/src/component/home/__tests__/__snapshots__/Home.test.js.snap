// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot for existing users 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-16"
  >
    <div
      class="MuiBox-root MuiBox-root-18"
    >
      <div
        class="MuiBox-root MuiBox-root-19"
      >
        <div
          class="MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              aria-label="disabled tabs example"
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Recent Projects
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-20 PrivateTabIndicator-colorPrimary-21 MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-25 makeStyles-buttonGroup-24"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add New
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-26 makeStyles-projectListContainer-17"
      >
        <div
          aria-labelledby="simple-tab-0"
          id="simple-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-grid-xl-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
            >
              <div
                class="MuiBox-root MuiBox-root-28 makeStyles-cardContainer-27"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  project1
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home should match snapshot for new user 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-mainContainer-4"
    >
      <h4
        class="MuiTypography-root makeStyles-textCentered-5 MuiTypography-h4"
      >
        Welcome to BharatSim
      </h4>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-itemsCentered-7"
      >
        <img
          alt="logo"
          src="welcomePlaceholder.svg"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-introTextContainer-6"
      >
        <h6
          class="MuiTypography-root makeStyles-textCentered-5 MuiTypography-subtitle2"
        >
          Intro text
        </h6>
        <p
          class="MuiTypography-root makeStyles-textCentered-5 MuiTypography-body2"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-itemsCentered-7"
      >
        <div
          class="MuiBox-root MuiBox-root-13 makeStyles-buttonGroup-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create new Project
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home should match snapshot for new user 2`] = `<div />`;
